routes/__init__.py

```diff
@@ -240,6 +241,72 @@ def mastodon_login_step2(instance_url):
     return resp
 
 
+@app.route('/login/misskey', methods=('GET', 'POST'))
+def misskey_login(instance=None):
+    instance_url = (request.args.get('instance_url', None)
+                    or request.form.get('instance_url', None))
+    
+    if not instance_url:
+        instances = libforget.misskey.suggested_instances(
+            limit = 30,
+            min_popularity = 1
+        )
+        return render_template(
+                'mastodon_login.html', instances=instances,
+                address_error=request.method == 'POST',
+                generic_error='error' in request.args
+                )
+
+    instance_url = instance_url.lower()
+    # strip protocol
+    instance_url = re.sub('^https?://', '', instance_url,
+                          count=1, flags=re.IGNORECASE)
+    # strip username
+    instance_url = instance_url.split("@")[-1]
+    # strip trailing path
+    instance_url = instance_url.split('/')[0]
```

Comment:
i can see a bug in this (a bug which i introduced) and honestly it would be wiser to make a single function using urllib.parse to extract the domain from a url, used both for mastodon and for misskey

up to you if you want to do that, otherwise i will

Reply:
Am now using `urllib.parse.urlparse` for this, and then taking the `netloc` part. This could technically contain a port too, but I think that should be fine? If somebody actually runs the instance on a non-standard port it might even be necessary. And it did not get filtered out before either.

Reply:
Yeah the port stuff is fine and, as you say, potentially necessary. I suppose we could remove the port if it matches the default port of the detected protocol, but given that protocol probing is not done here, I'm not sure it's worth the trouble.